alf/utils/dist_utils.py

```diff
@@ -560,6 +560,23 @@ def _get_categorical_builder(obj: td.Categorical):
         return td.Categorical, {'logits': obj.logits}
 
 
+def _get_onehot_categorical_builder(obj: td.OneHotCategorical):
+    if 'probs' in obj.__dict__ and id(obj.probs) == id(obj._param):
+        # This means that obj is constructed using probs
+        return td.OneHotCategorical, {'probs': obj.probs}
+    else:
+        return td.OneHotCategorical, {'logits': obj.logits}
+
+
+def _get_onehot_categorical_st_builder(
+        obj: td.OneHotCategoricalStraightThrough):
+    if 'probs' in obj.__dict__ and id(obj.probs) == id(obj._param):
+        # This means that obj is constructed using probs
+        return td.OneHotCategoricalStraightThrough, {'probs': obj.probs}
+    else:
+        return td.OneHotCategoricalStraightThrough, {'logits': obj.logits}
```

Comment:
You can probably merge the three functions above.

Reply:
Good point. Merged.